### DISCUSSION TITLE
No module named 'nonebot'启动不来提示没有nonebot模块，我已经安装了,求

### DISCUSSION THREAD

**Discussion by fjf199907** [nonebot/discussions#160] [问答]
提示没有nonebot模块，我已经安装了
Traceback (most recent call last):
  File "bot.py", line 1, in <module>
    import nonebot
ModuleNotFoundError: No module named 'nonebot'

---------------------------------------------------------------
import nonebot
from nonebot.adapters.onebot.v11 import Adapter
from services.db_context import init, disconnect

nonebot.init()
driver = nonebot.get_driver()
driver.register_adapter(Adapter)
config = driver.config
driver.on_startup(init)
driver.on_shutdown(disconnect)
# 优先加载定时任务
nonebot.load_plugin("nonebot_plugin_apscheduler")
nonebot.load_plugins("basic_plugins")
nonebot.load_plugins("plugins")
nonebot.load_plugins("extensive_plugin")
# 最后加载权限控制
nonebot.load_plugins("basic_plugins/hooks")

if __name__ == "__main__":
    nonebot.run()

**Comment by MingxuanGame** [comment]
检查你是否进入了虚拟环境

```bash
poetry shell
```

**Reply by fjf199907** [reply]
进入了
![INONTX42X2E10{3_}%`YK 1](https://user-images.githubusercontent.com/49164645/171543721-481b6796-1ba2-400b-add7-b072f30cf9ac.png)
--------------------------------------------
zhenxun-bot-UH4sDRIh-py3.10) root@fjf-virtual-machine:/home/zhenxun_bot# poetry shell
Virtual environment already activated: /root/.cache/pypoetry/virtualenvs/zhenxun-bot-UH4sDRIh-py3.10
(zhenxun-bot-UH4sDRIh-py3.10) root@fjf-virtual-machine:/home/zhenxun_bot# python bot.py
Traceback (most recent call last):
  File "/home/zhenxun_bot/bot.py", line 1, in <module>
    import nonebot
ModuleNotFoundError: No module named 'nonebot'
(zhenxun-bot-UH4sDRIh-py3.10) root@fjf-virtual-machine:/home/zhenxun_bot#
